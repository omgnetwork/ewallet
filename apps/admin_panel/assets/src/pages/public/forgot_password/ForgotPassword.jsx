import React, { Component } from 'react';
import { connect } from 'react-redux';

import ForgotPasswordForm from './ForgotPasswordForm';
import ForgotPasswordSuccess from './ForgotPasswordSuccess';

class ForgotPassword extends Component {
  constructor() {
    super();
    this.state = { didReset: false };
  }

  render() {
    const { didReset } = this.state;
    return (
      (didReset && <ForgotPasswordSuccess />) ||
      <ForgotPasswordForm onSuccess={() => { this.setState({ didReset: true }); }} />
    );
  }
}

export default connect()(ForgotPassword);
