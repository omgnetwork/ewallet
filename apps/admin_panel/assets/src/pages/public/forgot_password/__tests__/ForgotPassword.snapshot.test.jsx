import renderer from 'react-test-renderer';
import ForgotPassword from '../ForgotPassword';
import { SnapshotComponentTester } from '../../../../../__tests__/ComponentTester';

it('renders correctly', () => {
  const tree = renderer.create(SnapshotComponentTester(ForgotPassword, {})).toJSON();
  expect(tree).toMatchSnapshot();
});
