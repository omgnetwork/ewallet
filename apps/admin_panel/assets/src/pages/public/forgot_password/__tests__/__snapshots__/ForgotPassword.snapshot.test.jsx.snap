// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="omg-form"
>
  <h2
    className="omg-form__title"
  >
    Forgot your password?
  </h2>
  <h3
    className="omg-form__subtitle"
  >
    Enter your email address to reset your password.
  </h3>
  <form
    autoComplete="off"
    onSubmit={[Function]}
  >
    <div
      className="omg-form__group form-group"
    >
      <label
        className="omg-form__label control-label"
        htmlFor="email"
      >
        Email
      </label>
      <input
        className="omg-form__input form-control"
        id="email"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div>
      <span>
        <button
          className="btn btn-omg-blue btn btn-omg-blue-primary"
          disabled={true}
          onClick={[Function]}
          type="submit"
        >
          Reset your password
        </button>
        <span
          className="ml-1"
        >
          or
        </span>
        <button
          className="link-omg-blue btn btn-link"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Sign in
        </button>
      </span>
    </div>
  </form>
</div>
`;
